openapi/components/schemas/KycDocument/ProofOfIdentity.yaml

```
@@ -1,58 +1,29 @@
 allOf:
   - $ref: ../Common/CommonKycDocument.yaml
   - properties:
-      documentMatches:
+      customerInfo:
         type: object
         readOnly: true
         properties:
-          score:
-            description: >-
-              The calculated score that represents the % of confidence that this ID
-              represents the given customer.
-            type: number
-            format: double
-            example: 0.75
-          data:
-            type: object
-            properties:
-              containsImage:
-                description: Flag that indicates if there is an image that contains a face on it.
-                type: boolean
-                example: true
-              isIdentityDocument:
-                description: Flag that indicates if this looks like and ID.
-                type: boolean
-                example: true
-              isPublishedOnline:
-                description: If there is an exact match found online.
-                type: boolean
-                example: false
-              firstName:
-                description: The customer first name if it was matched, null otherwise.
-                type: string
-                example: John
-              lastName:
-                description: The customer last name if it was matched, null otherwise.
-                type: string
-                example: Doe
-              dateOfBirth:
-                description: The date of birth found on the document, null if not found.
-                allOf:
-                  - $ref: ../ServerTimestamp.yaml
-              expiryDate:
-                description: The expiry date found on the document, null if not found.
-                allOf:
-                  - $ref: ../ServerTimestamp.yaml
-              issueDate:
-                description: The issued date found on the document, null if not found.
-                allOf:
-                  - $ref: ../ServerTimestamp.yaml
-              hasMinimalAge:
-                description: >-
-                  Checks the minimal age, 21+ for USA and 18+ for all other countries.
-                  Null if dateOfBirth could not be determined.
-                type: boolean
-                example: true
+          firstName:
+            description: The customer first name if it was matched, null otherwise.
+            type: string
+            example: John
+          lastName:
+            description: The customer last name if it was matched, null otherwise.
+            type: string
+            example: Doe
+          dateOfBirth:
+            description: The date of birth found on the document, null if not found.
+            allOf:
+              - $ref: ../ServerTimestamp.yaml
+      documentMatches:
+        readOnly: true
+        allOf:
+          - $ref: ./IdentityMatches.yaml
+      overwriteData:
```

```suggestion
      documentMatches:
        allOf:
          - $ref: ./IdentityMatches.yaml
      parsedData:
        readOnly: true
```

I propose a minor change. They can edit the `documentMatches`. They have code that reads that and that's what they want to use. Instead, they can find the original data in the `parsedData` object.

Another possibility, is this should actually be a separate API:
```
POST /kyc-documents/{id}/documentMatches
{
   "firstName": "Adam",
   "lastName": "Altman", etc...

}

```

---

I like your proposition of the parsedData. As for the seperate API, would this eliminate the documentMatches in the GetKycDocument endpoint, or does it remain as it is, and this endpoint is added for updating?

---

It would remain as-is. This other API is exclusively for updating document matches data.